src/rdkafka_cgrp.c

```diff
@@ -3820,7 +3868,9 @@ rd_kafka_error_t *rd_kafka_consumer_group_metadata_read (
                         "Input buffer is not a serialized "
                         "consumer group metadata object");
 
-        group_id = buf + magic_len;
+        memcpy(&generation_id, buf+magic_len, sizeof(generation_id));
```

Comment:
This copy is unsafe, you haven't checked the length yet.
And why is it using sizeof(generation_id) instead of generationid_len? Pick one.

Reply:
whoops-a-daisy